commit 508514ed25315dd28340000831b17535d6f772da
Author: Dave Martin <dave.martin@linaro.org>
Date:   Mon Sep 3 13:49:25 2012 +0100

    ARM: 7511/1: opcodes: Opcode definitions for the Virtualization Extensions
    
    For now, this patch just adds a definition for the HVC instruction.
    More can be added here later, as needed.
    
    Now that we have a real example of how to use the opcode injection
    macros properly, this patch also adds a cross-reference from the
    explanation in opcodes.h (since without an example, figuring out
    how to use the macros is not that easy).
    
    Signed-off-by: Dave Martin <dave.martin@linaro.org>
    Acked-by: Nicolas Pitre <nico@linaro.org>
    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

diff --git a/arch/arm/include/asm/opcodes.h b/arch/arm/include/asm/opcodes.h
index f7937e1c46bf..74e211a6fb24 100644
--- a/arch/arm/include/asm/opcodes.h
+++ b/arch/arm/include/asm/opcodes.h
@@ -193,6 +193,8 @@ extern asmlinkage unsigned int arm_check_condition(u32 opcode, u32 psr);
  * specify the ARM and Thumb alternatives at the same time.  This ensures
  * that the correct opcode gets emitted depending on the instruction set
  * used for the kernel build.
+ *
+ * Look at opcodes-virt.h for an example of how to use these macros.
  */
 #include <linux/stringify.h>